garrysmod/lua/includes/modules/player_manager.lua

```diff
@@ -361,8 +373,11 @@ local function LookupPlayerClass( ply )
 	--
 	-- Check the cache
 	--
-	local method = ply.m_CurrentPlayerClass;
-	if ( method && method.Player == ply && method.ClassID == id && method.Func ) then return method end
+	local method = ply.m_CurrentPlayerClass
+	if ( method && method.Player == ply ) then
+		if ( method.ClassID == id && method.Func ) then return method end -- current class is still good, behave normally
+		if ( method.Destroy ) then method:Destroy() end -- the class id changed, destroy the old class

```

Comment:
Do you use PLAYER:Destroy to clean up player data on class change?

Reply:
Yes, some examples include removing clientside models, projected textures, and resetting bone manipulations.